src/Tools/ExternalAccess/Format/CodeStyleAnalyzers.cs

```diff
@@ -0,0 +1,223 @@
+                // - csharp_style_expression_bodied_lambdas
+                // - csharp_style_expression_bodied_local_functions
+                (new UseExpressionBodyDiagnosticAnalyzer(), new UseExpressionBodyCodeFixProvider()),
+
+                // Pattern matching
+                // - csharp_style_pattern_matching_over_is_with_cast_check
+                (new CSharpIsAndCastCheckDiagnosticAnalyzer(), new CSharpIsAndCastCheckCodeFixProvider()),
+                // - csharp_style_pattern_matching_over_as_with_null_check
+                (new CSharpAsAndNullCheckDiagnosticAnalyzer(), new CSharpAsAndNullCheckCodeFixProvider()),
+                // - csharp_style_prefer_switch_expression
+                (new ConvertSwitchStatementToExpressionDiagnosticAnalyzer(), new ConvertSwitchStatementToExpressionCodeFixProvider()),
+
+                // Inlined variable declarations
+                // - csharp_style_inlined_variable_declaration
+                (new CSharpInlineDeclarationDiagnosticAnalyzer(), new CSharpInlineDeclarationCodeFixProvider()),
+
+                // Expression-level preferences
+                // - csharp_prefer_simple_default_expression
+                (new CSharpUseDefaultLiteralDiagnosticAnalyzer(), new CSharpUseDefaultLiteralCodeFixProvider()),
+                // - csharp_style_deconstructed_variable_declaration
+                (new CSharpUseDeconstructionDiagnosticAnalyzer(), new CSharpUseDeconstructionCodeFixProvider()),
+                // - csharp_style_pattern_local_over_anonymous_function
+                (new CSharpUseLocalFunctionDiagnosticAnalyzer(), new CSharpUseLocalFunctionCodeFixProvider()),
+                // - csharp_style_prefer_index_operator
+                (new CSharpUseIndexOperatorDiagnosticAnalyzer(), new CSharpUseIndexOperatorCodeFixProvider()),
+                // - csharp_style_prefer_range_operator
+                (new CSharpUseRangeOperatorDiagnosticAnalyzer(), new CSharpUseRangeOperatorCodeFixProvider()),
+                // - csharp_prefer_static_local_function
+                (new MakeLocalFunctionStaticDiagnosticAnalyzer(), new MakeLocalFunctionStaticCodeFixProvider()),
+                // - csharp_prefer_simple_using_statement
+                (new UseSimpleUsingStatementDiagnosticAnalyzer(), new UseSimpleUsingStatementCodeFixProvider()),
+
+                // "Null" checking preferences
+                // - csharp_style_throw_expression
+                (new CSharpUseThrowExpressionDiagnosticAnalyzer(), new UseThrowExpressionCodeFixProvider()),
+                // - csharp_style_conditional_delegate_call
+                (new InvokeDelegateWithConditionalAccessAnalyzer(), new InvokeDelegateWithConditionalAccessCodeFixProvider()),
+
+                // Code block preferences
+                // - csharp_prefer_braces
+                (new CSharpAddBracesDiagnosticAnalyzer(), new CSharpAddBracesCodeFixProvider()),
+            }.ToImmutableArray();
```

Comment:
this honestly just makes me sad.  this is like the 4th time that we've gone and just hardcoded a list of these things in totally random places across the codebase.  Why not just have a discovery mechanism where these features can register and say they'll be avialable, and then that can be discovered and exported using a totally supported system like MEF?

Reply:
DiagnosticAnalyzers can't be registered with MEF because there is no MEF in command line compilation scenarios.